RELEASE.md

```diff
@@ -5,7 +5,8 @@
     * A curated list of notable changes and links to PRs. Do not simply dump git log, select the changes that affect the users.
       To obtain the list of all changes run `make changelog` or use `scripts/release-notes.py`.
     * The section can be split into sub-section if necessary, e.g. UI Changes, Backend Changes, Bug Fixes, etc.
-    * If the jaeger-ui submodule has changes cut a new release and also upgrade the submodule versions then commit, for example:
+    * If the jaeger-ui submodule has changes cut a new release
```

Comment:
We agreed that UI release numbers will now match the backend release numbers, so "if ui has changes" is misleading, we should make it unconditional Step 1: release UI.